### PR TITLE
Run multi-module check during presubmit

### LINKED ISSUE #2774: Pre-submit should fail on PRs that span modules.
A PR with one or more commits that spans multiple Go modules in this repo makes it harder to understand the actual changes being delivered in a module release (see #2773).

[Makefile]: https://github.com/kubernetes-sigs/kustomize/blob/master/Makefile

Currently the presubmit does the equivalent of 
```
make prow-presubmit-check 
```
(see comment above that target in the [Makefile]).

Lets add check there that fails a PR which contains commits which span modules.

An admin can merge a PR regardless, so this doesn't block emergency fixes.

Also, arrange things so that if a PR author adds
```
MODULE_SPAN_OK
```
to the PR description, it disables the check, so that other tests still run.

### PR DESCRIPTION
**Pull request by runewake2**
Runs multi-module prcheck script during prow presubmit.

This will flag PR's with commits that make changes to two or more of the following paths simultaneously:

```
cmd/config
api/
kustomize/
kyaml/
```

Fixes #2774

### CODE REVIEW

File: cmd/prchecker/main.go [hunk 1]
```diff
@@ -237,7 +250,7 @@ func PullRequestSpanningPathList(repository GitHubService, pullrequest int, path
 	return spanningChangesExist, changesets, nil
 }
 
-// IsChangesetSpanningPathList tests if a changeset is spanning
+// ChangesetSpanningPathList tests if a changeset is spanning
```
Comment (monopole) [inline cmd/prchecker/main.go:253]: - please make this a method on Changeset.
 - name it so it reads well in an `if` statement, e.g.

```
if a.equals(b) {}
if a.contains(b) {}
if changeSet.spansPaths(paths) {}
```

Comment (runewake2) [reply cmd/prchecker/main.go]: Renamed to `isSpanningPaths()`

Comment (monopole) [reply cmd/prchecker/main.go]: OK, but fwiw, there are plenty of Go functions in the standard libs that return bool that don't begin with `is`;
https://golang.org/pkg/path/filepath/#HasPrefix
https://golang.org/pkg/strconv/#CanBackquote
https://golang.org/pkg/time/#Time.After
etc.

**Comment by monopole** [comment]
Test is failing.  

Suggest taking the Makefile changes out of this PR and into a new one to debug it.

Then this pr can be just the changes to prchecker.

**Comment by runewake2** [comment]
It's an issue with https://github.com/kubernetes-sigs/kustomize/pull/3187/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R235

Taking a quick look.

File: Makefile [hunk 2]
```diff
@@ -226,6 +228,16 @@ test-unit-cmd-all:
 test-go-mod:
 	./travis/check-go-mod.sh
 
+# Environment variables are defined at
+# https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables
+.PHONY: test-multi-module
+test-multi-module: $(MYGOBIN)/prchecker
+	cd ./cmd/prchecker; go run . \
```
Comment (monopole) [inline Makefile:235]: just
```
$(MYGOBIN)/prchecker \
```
It will be built by L234 and the associated dependency target

**Pull request merged**